### PR TITLE
propose HIPE: Anoncreds protocol

### PR DESCRIPTION
**Pull request by brentzundel**
This HIPE is a description of the cryptography used in the anoncreds protocol as it is currently implemented.

### CODE REVIEW

File: text/anoncreds-protocol/README.md [hunk 1]
```diff
@@ -0,0 +1,624 @@
...
+about their identity an uncorrelatable way without revealing other identity
+details. The properties can be raw identity attributes such as a birth date or
+address, or more sophisticated predicates such as "A is older than 20 years
+old".
+
+We assume three parties: *issuer*, *holder*, and *verifier*. From the functional
+perspective:
+* the issuer gives a credential *C* based on identity schema *X*,
+which asserts certain properties 𝒫 about *X*, to the holder.
+* The credential consists of attributes represented by integers
```
Comment (jovfer) [inline text/anoncreds-protocol/README.md]: Should this HIPE reference or cover topic around user-friendly attributes and integers for math?

Comment (brentzundel) [reply text/anoncreds-protocol/README.md]: something like how the sdk encodes the user-friendly attributes into integers?

Comment (jovfer) [reply text/anoncreds-protocol/README.md]: yes. In my mind this HIPE should at least mention, that we have 2 representation of attributes and math operates only with integers. 
and libindy allow to use application encoding (from initial raw representation to integer) and doesn't check anything vs libvcx use hashing to automatically encode non-integer attribs.

I'm not sure should this HIPE cover a lot of details, but I think we have to describe at least two types and point out, that mathematics proof can confirm only integers and it's up to application/library above libursa & libindy to check validity of encoding.

Comment (brentzundel) [reply text/anoncreds-protocol/README.md]: I've added a section in [d464b93](https://github.com/hyperledger/indy-hipe/pull/109/commits/d464b933254f17068ea70ce5b97d43ca7e86e710). Please let me know if it addresses your concerns, or if you feel more should be added.

**Comment by dhh1128** [comment]
@brentzundel and @ken-ebert : I have raised a PR against your PR to add a HIPE number and status section to your README.md. Please merge my PR. Once you do, this PR can be merged. We are now following a streamlined process; see the updated description in the repo's /README.md.

**Pull request merged**